chore(app-shell-odd): update electron builder config

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
Update electron in builder config file.
<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Test Plan

<!--
Use this section to describe the steps that you took to test your Pull Request.
If you did not perform any testing provide justification why.

OT-3 Developers: You should default to testing on actual physical hardware.
Once again, if you did not perform testing against hardware, justify why.

Note: It can be helpful to write a test plan before doing development

Example Test Plan (HTTP API Change)

- Verified that new optional argument `dance-party` causes the robot to flash its lights, move the pipettes,
then home.
- Verified that when you omit the `dance-party` option the robot homes normally
- Added protocol that uses `dance-party` argument to G-Code Testing Suite
- Ran protocol that did not use `dance-party` argument and everything was successful
- Added unit tests to validate that changes to pydantic model are correct

-->

# Changelog

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
chore(app-shell-odd): update electron version info in build config file

update electron version info in build config file

koji committed:
chore(app-shell-odd): update electron version info in build config file

update electron version info in build config file

koji committed:
add app.getVersion to retrieve electron version

koji committed:
add app.getVersion to retrieve electron version

shlokamin commented on app-shell-odd/electron-builder.config.js:
```diff
@@ -1,8 +1,9 @@
 module.exports = {
   appId: 'com.opentrons.odd',
-  electronVersion: '23.3.13',
+  electronVersion: app.getVersion(),
```
i dont think this is quite right: this will get the version of the app, not electron: https://www.electronjs.org/docs/latest/api/app#appgetversion

i think we want to import our `package.json` directly: 

```js
import pkg from './package.json'

module.exports = {
...
  electronVersion: JSON.stringify(pkg.devDependencies.electron),
}
```

koji committed:
address comments

koji committed:
address comments

shlokamin reviewed (approved):
lgtm but lets put this in edge instead of the the release branch to minimize variables since they want to release ASAP

also maybe lets wait to merge this into edge until the vite branch is merged?

sfoster1 commented:
Corresponding openembedded branch: https://github.com/Opentrons/oe-core/pull/137

y3rsh reviewed (commented):
Along with this would we also want to dynamically define electron version in /app-shell/Makefile ?  Not sure off the top of my head if that is easy to do in a reliable way. 
https://github.com/Opentrons/opentrons/blob/release/app-shell/Makefile#L62

koji committed:
remove pkg

koji committed:
pass the electron version directly

sfoster1 commented:
With this build https://github.com/Opentrons/oe-core/actions/runs/8238878130 of the oe-core PR and with putting all the built in solibs in, the app doesn't haeve a crash and all its processes are running but is still not displaying

sfoster1 committed:
show mainwindow after first browser dispatch

We used to show mainWindow on the ready-to-show event. For some reason
that's just not happening anymore, but everything underneath is actually
working fine. Instead, we can show when we get the first dispatch from
the app side. This may lead to us spending more time on the pre-app
splash and less tim on the spinner.

sfoster1 commented:
Latest commit alters init behavior. Build with it is here https://github.com/Opentrons/oe-core/actions/runs/8332543203

That openembedded PR will definitely be needed because it also now adds a command line flag that tells chrome to allow all-origin requests for devtools websockets.

sfoster1 commented:
> Along with this would we also want to dynamically define electron version in /app-shell/Makefile ? Not sure off the top of my head if that is easy to do in a reliable way. https://github.com/Opentrons/opentrons/blob/release/app-shell/Makefile#L62

I'm just noting this and uhhhhhhhhhhh what

sfoster1 committed:
add some todos

sfoster1 commented:
> Along with this would we also want to dynamically define electron version in /app-shell/Makefile ? Not sure off the top of my head if that is easy to do in a reliable way. https://github.com/Opentrons/opentrons/blob/release/app-shell/Makefile#L62

OK, had some time to think about this. I think that we should, but maybe in a follow up pr, and honestly I think the most reliable thing to do is use `jq`.

I think we _do not_ want to do this for the ODD. There, we just accept that the version might be different than the desktop.
